### PR TITLE
Strip /scm/ prefix from Bitbucket SSH URLs to fix REST API path resolution

### PR DESCRIPTION
**Pull request by gnh374**
## Summary
  - Fix SSH URL regex to strip optional `/scm/` prefix from Bitbucket Server SSH URLs
  - Escape `.` before `git` in regex to match literal dot only

  ## Problem
  Bitbucket Server SSH URLs may contain `/scm/` in the path (e.g., `ssh://git@host/scm/project/repo.git`). The previous regex captured `scm/` as part of the project key, resulting
  in malformed REST API calls like `/rest/api/1.0/projects/scm/project/repos/repo/commits`. Bitbucket returns an HTML error page for this invalid path, which Feign fails to parse as
   JSON:

  Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')

  ## Fix
  Changed the SSH URL regex from:
  $sshUrl[:/]((?:[^/]+/)+)([^/]+).git
  To:
  $sshUrl[:/](?:scm/)?((?:[^/]+/)+)([^/]+)\.git

  `(?:scm/)?` optionally matches and discards the `/scm/` segment before capture groups. This fixes all endpoints that rely on `parse()` (`getIssuesFromCommits`, `getCommits`,
  `getTags`, etc.). URLs without `/scm/` are unaffected. Gitea URLs are unaffected.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Git SSH URL recognition and parsing: stricter host matching, literal ".git" enforcement, and optional intermediate path segment limited to the Bitbucket format to reduce false matches.
* **Tests**
  * Added unit tests validating SSH URL parsing and support-detection, including edge cases for host matching and presence/absence of the intermediate path segment.

### CODE REVIEW

File: server/src/main/kotlin/org/octopusden/octopus/vcsfacade/service/VcsService.kt [hunk 1]
```diff
@@ -18,7 +18,7 @@ abstract class VcsService(vcsServiceProperties: VcsProperties.Service) {
     val indexing = vcsServiceProperties.indexing
     protected val httpUrl = vcsServiceProperties.httpUrl.lowercase().trimEnd('/')
     protected val sshUrl = vcsServiceProperties.sshUrl.lowercase().trimEnd(':', '/')
-    private val sshUrlRegex = "$sshUrl[:/]((?:[^/]+/)+)([^/]+).git".toRegex()
+    private val sshUrlRegex = "$sshUrl[:/](?:scm/)?((?:[^/]+/)+)([^/]+)\\.git".toRegex()
```
[the remark this answers is not in the document]
Comment (adellinekania) [reply server/src/main/kotlin/org/octopusden/octopus/vcsfacade/service/VcsService.kt]: @gnh374, please check this & ensure the build is green

[the remark this answers is not in the document]
Comment (gnh374) [reply server/src/main/kotlin/org/octopusden/octopus/vcsfacade/service/VcsService.kt]: > @gnh374, please check this & ensure the build is green

fixed

**Comment by pgorbachev** [comment]
Could we add a small unit test for `parse` / `isSupported` as part of this PR? This is the second regex fix in this class, and the parser is on the hot path for every REST call, so a parametric test covering something like:

- `ssh://host/scm/proj/repo.git` → `(proj, repo)` — normal Bitbucket Server URL
- `ssh://host/scm/SCM/repo.git` / `ssh://host/scm/scm/repo.git` → `(scm, repo)` — Bitbucket project literally keyed `SCM` (the case that motivated this fix in the first place)
- `ssh://host/proj/repo.git` → `(proj, repo)` — non-Bitbucket / no route prefix
- `ssh://host/proj/sub/repo.git` → `(proj/sub, repo)` — nested groups
- `ssh://host/proj/xgit` → not matched (regression guard for the `.` → `\.` change)

would help us avoid a third round of fixes here later.

**Comment by pgorbachev** [comment]
One thing on scope: `/scm/` is a Bitbucket Server routing quirk, but this change lives in the base `VcsService`, so it now affects Gitea too. A Gitea org/project literally named `scm` would be silently stripped during parsing. Very unlikely in practice, but it feels like this logic should either be overridden in `BitbucketService`, or at least gated by `type == BITBUCKET`. Not a blocker for me — just flagging.

**Comment by gnh374** [comment]
> One thing on scope: `/scm/` is a Bitbucket Server routing quirk, but this change lives in the base `VcsService`, so it now affects Gitea too. A Gitea org/project literally named `scm` would be silently stripped during parsing. Very unlikely in practice, but it feels like this logic should either be overridden in `BitbucketService`, or at least gated by `type == BITBUCKET`. Not a blocker for me — just flagging.

fixed

**Comment by gnh374** [comment]
> Could we add a small unit test for `parse` / `isSupported` as part of this PR? This is the second regex fix in this class, and the parser is on the hot path for every REST call, so a parametric test covering something like:
> 
> * `ssh://host/scm/proj/repo.git` → `(proj, repo)` — normal Bitbucket Server URL
> * `ssh://host/scm/SCM/repo.git` / `ssh://host/scm/scm/repo.git` → `(scm, repo)` — Bitbucket project literally keyed `SCM` (the case that motivated this fix in the first place)
> * `ssh://host/proj/repo.git` → `(proj, repo)` — non-Bitbucket / no route prefix
> * `ssh://host/proj/sub/repo.git` → `(proj/sub, repo)` — nested groups
> * `ssh://host/proj/xgit` → not matched (regression guard for the `.` → `\.` change)
> 
> would help us avoid a third round of fixes here later.

added

**Review by adellinekania — APPROVED** [review]

**Review by rodemark — APPROVED** [review]
